### ISSUE TITLE
password is read from stdin, not TTY

### ISSUE THREAD

**Issue by dw** [slicer69/doas#17]
repro:   echo foo | doas cat

This breaks piping something to a command via doas, and breaks tools (dw/mitogen) expecting it to behave like the BSD version

**Comment by slicer69** [comment]
I've tested this and it works as expected on FreeBSD.

On Linux it was echoing the password when a pipe was used, which is obviously not good. I've fixed that.

On Linux piped data is not being forwarded to the command run by doas, it is intercepted by doas. Will look into that further.

**Comment by psposito** [comment]
Is there any progress about this, as mentioned above?

**Comment by slicer69** [comment]
No, and there probably won't be until someone submits a patch for Linux compatibility that doesn't break other platforms.

**Comment by sandsmark** [comment]
it doesn't seem like this does any tty or pty handling at all?

This is the basic stuff needed for securely reading the password AFAIK: https://github.com/sudo-project/sudo/blob/9e111eae57524ca72002ad1db36eb68ccd50b167/src/tgetpass.c#L110-L284

I don't think that is linux specific.

**Comment by slicer69** [comment]
This is Linux specific, the other platforms just handle this, basically automatically.

**Comment by henriquehbr** [comment]
Quoting my suggested solution mentioned on https://github.com/Duncaen/OpenDoas/issues/21

> It turns out i found a pretty good workaround for this by using `expect` for automatically entering input on interactive prompts, i ended up with something like this:
> ```bash
> #!/usr/bin/expect
> spawn doas -- pacman --noconfirm -U yay-bin-10.2.3-1-x86_64.pkg.tar.zst
> expect "Password: " {send -- "password\r"}
> expect eof
> ```
> Hope it helps someone :)
